### PR TITLE
Convert from 32 bytes

### PR DESCRIPTION
**Pull request by benbierens**

### CODE REVIEW

File: poseidon2/io.nim [hunk 1]
```diff
@@ -27,15 +35,17 @@ iterator elements*(bytes: openArray[byte], _: type F): F =
...
   var chunkStart = 0
   while chunkStart + chunkLen <= bytes.len:
     let chunkEnd = chunkStart + chunkLen - 1
     let element = F.fromOpenArray(bytes.toOpenArray(chunkStart, chunkEnd))
     yield element
     chunkStart += chunkLen
-  let finalChunk = bytes[chunkStart..<bytes.len] & endMarker
-  let finalElement = F.fromOpenArray(finalChunk)
-  yield finalElement
+  if (bytes.len - chunkStart > 0) or empty:
```
Comment (dryajov) [inline poseidon2/io.nim]: I think this was pulled from my padding PR which ended up being the wrong fix. Did something change, or was this pulled in by mistake?

File: tests/poseidon2/testIo.nim [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import std/unittest
 import std/sequtils
+import std/options
```
Comment (emizzle) [inline tests/poseidon2/testIo.nim:3]: needed?

**Comment by dryajov** [comment]
Can we get this merged?

File: tests/poseidon2/testIo.nim (hunk 2 above)
Comment (benbierens) [reply tests/poseidon2/testIo.nim:3]: yes, is used :o

**Pull request merged**